src/platform/sys.c

```diff
@@ -1,17 +1,25 @@
 #include "moar.h"
 #include "platform/sys.h"
 
-MVMuint32 MVM_platform_cpu_count(void) {
-    int count;
+MVMint64 MVM_platform_cpu_count(void) {
+    MVMint64       count;
     uv_cpu_info_t *info;
-    int e;
+    int            e;
 
-    e = uv_cpu_info(&info, &count);
-    if (e == 0) uv_free_cpu_info(info, count);
+    e = uv_cpu_info(&info, (int *)(&count));
```

Comment:
Why was the type of `count` changed from `int`? This could cause problems on big endian, maybe? I'd change `count` back to `int` and then just have it upcast to `MVMint64` implicitly at return.

Reply:
I was naively thinking MoarVM only supported amd64 because of the legojit, which would've made `int` 64 bits. It was a dumb assumption on my part. I'll change it back

Reply:
We only JIT on that platform, but we try - with varying degrees of success - to run on others. :-)